### DISCUSSION TITLE
How to add hover, focus and active styling globally for lofi theme

### DISCUSSION THREAD

**Discussion by roshansachan** [saadeghi/daisyui#1692] [Ideas / feature requests]
I see that the buttons in the Lofi theme do not provide any feedback for hover, focus and active states. Is there a way I can enable/add this globally ?

**Comment by saadeghi** [comment]
If you mean the button click animation, you can change it from `tailwind.config.js` file like this:  
https://play.tailwindcss.com/IEg64g8P9q?file=config
```js
  daisyui: {
    themes: [
      {
        lofi: {
          ...require("daisyui/src/colors/themes")["[data-theme=lofi]"],
          "--animation-btn": ".25s",
          "--btn-focus-scale":".95",
        },
      },
    ],
  },
```

Or you can change it in CSS file like this:  
https://play.tailwindcss.com/bb0cOmSY79?file=css
```css
[data-theme=lofi]{
  --animation-btn: .25s;
  --btn-focus-scale: .95;
}
```

**Reply by roshansachan** [reply]
Thanks for the quick reply @saadeghi . This would only add an animation on focus state. Is there a way to change the background color/opacity  on hover/active like it is there in other themes?
I am trying to change the background and opacity of the button for hover/focus and active state. While I can add the css for this just like you mentioned above but I would have to hardcode the color values becuase I am not sure if I can use primary-focus variable/class in tailwind config.

 Attaching an example,

https://user-images.githubusercontent.com/4952904/225302577-a10870fd-be6d-4c8c-8dba-07079a0aaad2.mov

**Reply by saadeghi** [reply]
in `lofi` theme all brand colors (primary, secondary, accent) and their `*-focus` colors are black. because it's a "lo-fi" theme.  
But you can customize every color as you wish.  
You can also put any custom CSS in a theme config. See this example:  
https://play.tailwindcss.com/zKLI2f8mZZ?file=config

**Reply by roshansachan** [reply]
Perfect. Thanks.
